include/nfl/poly.hpp

```diff
@@ -151,6 +151,15 @@ class poly {
   void ntt_pow_phi() { base.ntt_pow_phi(*this);}
   void invntt_pow_invphi() { base.invntt_pow_invphi(*this); }
 
+  /* manual serializers
+  */
+  inline void serialize(std::ostream& outputstream) {
+    outputstream.write(reinterpret_cast<char*>(_data), N * sizeof(T));
+  }
+  inline void deserialize(std::istream& inputstream) {
+    inputstream.read(reinterpret_cast<char*>(_data), N * sizeof(T));
+  }
```

Comment:
> Would boost/libcereal be portable? 

They can be (at least for boost) is you choose a portable backend,
generally a textual one

> What should we do eventually? Do it manually, use boost (I couldn't get it to work with adding boost as a dependence), use libcereal? I was up for libcereal but then we discussed about doing it manually, and eventually it seems a problematic approach.

It really depends on the use case, and I don't know about them.
basically if the data are meant to be shared over the network, then
using a serialization library would avoid this issues.